pkg/gofr/datasource/pubsub/nats/interfaces.go

```diff
@@ -0,0 +1,70 @@
+package nats
+
+import (
+	"context"
+
+	"github.com/nats-io/nats.go"
+	"github.com/nats-io/nats.go/jetstream"
+
+	"gofr.dev/pkg/gofr/datasource"
+	"gofr.dev/pkg/gofr/datasource/pubsub"
+)
+
+//go:generate mockgen -destination=mock_client.go -package=nats -source=./interfaces.go Client,Subscription,ConnInterface,ConnectionManagerInterface,SubscriptionManagerInterface,StreamManagerInterface
+
+// ConnInterface represents the main Client connection.
+type ConnInterface interface {
+	Status() nats.Status
+	Close()
+	NATSConn() *nats.Conn
+	JetStream() (jetstream.JetStream, error)
+}
+
+// NATSConnector represents the main Client connection.
+type NATSConnector interface {
```

Comment:
```suggestion
type Connector interface {
```

You are in nats package

Reply:
Sure done